src/components/leaflet/LeafletMap.jsx

```diff
@@ -1,20 +1,29 @@
 import { MapContainer, TileLayer, Marker } from 'react-leaflet';
 import style from './leaflet.module.scss';
+import { Icon } from 'leaflet';
+import MarkerIcon from '../../assets/images/icon-location.svg';
+
+const customIcon = new Icon({
+  iconUrl: MarkerIcon,
+  iconSize: [42, 52],
+  iconAnchor: [16, 32],
+  popupAnchor: [0, -32],
+});
 
 const LeafletMap = (props) => {
   return (
     <MapContainer
+      key={`${props.coordinates[0]}`}
```

Comment:
Try something like this 
```js
const RecenterAutomatically = ({ lat, lng }) => {
  const map = useMap();
  const { lat: prevLat, lng: prevLng } = map.getCenter();
  const latDiff = Math.abs(prevLat - lat);
  const lngDiff = Math.abs(prevLng - lng);
  const largestDiff = Math.abs(latDiff - lngDiff);
  const duration = Math.ceil(largestDiff * 0.05);

  useEffect(() => {
    map.setView([lat, lng], 8, {
      animate: true,
      easeLinearity: 0.5,
      duration,
    });
  }, [lat, lng]);

  return null;
};

const LeafletMap = ({ coordinates }) => {
  return (
    <MapContainer className={style.mapCont} center={coordinates} zoom={8} scrollWheelZoom={false}>
      <TileLayer
        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
      />
      <Marker position={coordinates} icon={customIcon} />
      <RecenterAutomatically lat={coordinates[0]} lng={coordinates[1]} />
    </MapContainer>
  );
};
```

Reply:
By the way, you don't need to do this on this PR. We have a dedicated task for a sync between the map and ip data.